lib/posix/pthread_rwlock.c

```diff
@@ -155,9 +154,9 @@ int pthread_rwlock_timedwrlock(pthread_rwlock_t *rwlock,
 		return EINVAL;
 	}
 
-	timeout = (s32_t) timespec_to_timeoutms(abstime);
+	timeout = (s32_t) _timespec_to_timeoutms(abstime);
 
-	if (write_lock_acquire(rwlock, timeout) != 0U) {
+	if (timeout <= 0 || write_lock_acquire(rwlock, timeout) != 0U) {
```

Comment:
This is not correct: http://pubs.opengroup.org/onlinepubs/9699919799/functions/pthread_rwlock_timedwrlock.html

> Under no circumstances shall the function fail with a timeout if the lock can be acquired immediately. The validity of the abstime parameter need not be checked if the lock can be immediately acquired.